### PR TITLE
[ Feature ] Add support for GCP secrets manager

### PR DESCRIPTION
**Pull request by ZeljkoBenovic**
# Description

This PR adds the support for storing secrets in Google Cloud Platform secrets manager.    

A new `--type` option in the `secrets generate` command has been added - `gcp-ssm`.
   
When running `secrets generate` command the operator needs to specify two new parameters in `--extra` flag - `project-id` - the project id provided in GCP dashboard and `gcp-ssm-cred` - json file with credentials exported from GCP console

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have added the relevant labels
- [x] I have updated the official documentation
- [x] I have added sufficient documentation in code

## Testing

- [ ] I have tested this code with the official test suite
- [x] I have tested this code manually

### Manual tests

In order to test this, first of all you'll need to go to [GCP Console](https://console.cloud.google.com) and enable Billing Account for your Gmail account but  there is also free trial available if you're using GCP for the first time. 

Next, find and enable [Secrets Manager API](https://console.cloud.google.com/apis/library/browse) and get the credentials from GCP console by following these [instructions ](https://cloud.google.com/secret-manager/docs/reference/libraries)

Generate `secretsManagerConfig.json` file by running `polygon-edge secrets generate --name node1 --extra project-id=<your_project_id>,gcp-ssm-cred=<credentials_json_file> --type gcp-ssm`   
Store secrets in GCP SM by running `secrets init --config secretsManagerConfig.json`
You should now see the stored secrets in GCP Secrets Manager console.

When starting the server, among standard flags, we need to specify `--secrets-config secretsManagerConfig.json`

# Documentation update

Public documentation PR link: https://github.com/0xPolygon/polygon-edge-docs/pull/86

## Additional Comments
Fixes EDGE-508

### CODE REVIEW

**Comment by zivkovicmilos** [comment]
@ZeljkoBenovic 

Please don't forget to regenerate the licenses json in this PR 🙏

**Comment by ZeljkoBenovic** [comment]
Licenses updated

**Comment by Anny1730** [comment]
> # Description
> This PR adds the support for storing secrets in Google Cloud Platform secrets manager.
> 
> A new `--type` option in the `secrets generate` command has been added - `gcp-ssm`.
> 
> When running `secrets generate` command the operator needs to specify two new parameters in `--extra` flag - `project-id` - the project id provided in GCP dashboard and `gcp-ssm-cred` - json file with credentials exported from GCP console
> 
> # Changes include
> * [ ]  Bugfix (non-breaking change that solves an issue)
> * [ ]  Hotfix (change that solves an urgent issue, and requires immediate attention)
> * [x]  New feature (non-breaking change that adds functionality)
> * [ ]  Breaking change (change that is not backwards-compatible and/or changes current functionality)
> 
> # Checklist
> * [x]  I have assigned this PR to myself
> * [x]  I have added at least 1 reviewer
> * [x]  I have added the relevant labels
> * [x]  I have updated the official documentation
> * [x]  I have added sufficient documentation in code
> 
> ## Testing
> * [ ]  I have tested this code with the official test suite
> * [x]  I have tested this code manually
> 
> ### Manual tests
> In order to test this, first of all you'll need to go to [GCP Console](https://console.cloud.google.com) and enable Billing Account for your Gmail account but there is also free trial available if you're using GCP for the first time.
> 
> Next, find and enable [Secrets Manager API](https://console.cloud.google.com/apis/library/browse) and get the credentials from GCP console by following these [instructions ](https://cloud.google.com/secret-manager/docs/reference/libraries)
> 
> Generate `secretsManagerConfig.json` file by running `polygon-edge secrets generate --name node1 --extra project-id=<your_project_id>,gcp-ssm-cred=<credentials_json_file> --type gcp-ssm` Store secrets in GCP SM by running `secrets init --config secretsManagerConfig.json` You should now see the stored secrets in GCP Secrets Manager console.
> 
> When starting the server, among standard flags, we need to specify `--secrets-config secretsManagerConfig.json`
> 
> # Documentation update
> Public documentation link will be provided soon
> 
> ## Additional Comments
> Fixes EDGE-508

**Comment by Anny1730** [comment]
Repo

File: command/secrets/generate/secrets_generate.go [hunk 1]
```diff
@@ -51,29 +49,26 @@ func setFlags(cmd *cobra.Command) {
 		typeFlag,
 		string(secrets.HashicorpVault),
 		fmt.Sprintf(
-			"the type of the secrets manager. Default: %s",
+			"the type of the secrets manager. Available types: %s, %s, %s and %s",
+			secrets.Local,
```
Comment (zivkovicmilos) [inline command/secrets/generate/secrets_generate.go]: Please note that we don't use `secrets generate` for the local file system key management.
We use `secrets init` for that purpose, this generate command is only for generating the config

File: secrets/gcpssm/gcp_ssm.go [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+func SecretsManagerFactory(
+	config *secrets.SecretsManagerConfig,
+	params *secrets.SecretsManagerParams,
+) (secrets.SecretsManager, error) {
+	// Check if project id is defined
+	if _, ok := config.Extra["project-id"]; !ok {
+		return nil, errors.New("no project-id variable specified")
+	}
+	// Check if gcp-ssm-cred is defined
+	if _, ok := config.Extra["gcp-ssm-cred"]; !ok {
```
Comment (zivkovicmilos) [inline secrets/gcpssm/gcp_ssm.go]: Can we make these `Extra` fields constants of the package?
It seems cleaner to extract them out

File: secrets/gcpssm/gcp_ssm.go [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+		return nil, errors.New("no gcp-ssm-cred variable specified")
+	}
+	// Check if the variables are present
+	if config.Name == "" || config.Extra["project-id"] == "" || config.Extra["gcp-ssm-cred"] == "" {
+		return nil, errors.New("name, project-id or gcp-ssm-cred can't be empty string")
```
Comment (zivkovicmilos) [inline secrets/gcpssm/gcp_ssm.go]: Typo here, `can't be an empty string`

File: secrets/gcpssm/gcp_ssm.go [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+	"fmt"
+	"github.com/0xPolygon/polygon-edge/secrets"
+	"github.com/hashicorp/go-hclog"
+	"os"
+
+	sm "cloud.google.com/go/secretmanager/apiv1"
+	smpb "google.golang.org/genproto/googleapis/cloud/secretmanager/v1"
+)
+
+type GcpSsmManager struct {
```
Comment (zivkovicmilos) [inline secrets/gcpssm/gcp_ssm.go]: Nitpick: What do you think about naming this `GCPSSMManager`?

Comment (zivkovicmilos) [reply secrets/gcpssm/gcp_ssm.go]: Or just `GCPSecretsManager`

**Review by zivkovicmilos — APPROVED** [review]
Looks great 💯

File: secrets/gcpssm/gcp_ssm.go [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+
+	return gcpSsmManager, nil
+}
+
+// Setup performs secret manager-specific setup
+func (gm *GcpSsmManager) Setup() error {
+	var clientErr error
+	// Set environment variable that specifies credentials.json file path
+	if err := os.Setenv(
+		"GOOGLE_APPLICATION_CREDENTIALS",
```
Comment (Kourin1996) [inline secrets/gcpssm/gcp_ssm.go]: Can you define constant value?

**Review by Kourin1996 — APPROVED** [review]
LGTM. That's nice! It works with my local machine.

File: command/secrets/generate/secrets_generate.go (hunk 1 above)
Comment (ZeljkoBenovic) [reply command/secrets/generate/secrets_generate.go]: Got it. Should we remove the support for `--type local` from the code completely ? It seams redundant...

File: secrets/gcpssm/gcp_ssm.go (hunk 2 above)
Comment (ZeljkoBenovic) [reply secrets/gcpssm/gcp_ssm.go]: Agreed. Much cleaner now b95eb9d2fcd7a213bc19ef0ab427cde103615642

File: secrets/gcpssm/gcp_ssm.go (hunk 3 above)
Comment (ZeljkoBenovic) [reply secrets/gcpssm/gcp_ssm.go]: Fixed in b95eb9d2fcd7a213bc19ef0ab427cde103615642

File: secrets/gcpssm/gcp_ssm.go (hunk 4 above)
Comment (ZeljkoBenovic) [reply secrets/gcpssm/gcp_ssm.go]: Renamed b95eb9d2fcd7a213bc19ef0ab427cde103615642

File: secrets/gcpssm/gcp_ssm.go (hunk 5 above)
Comment (ZeljkoBenovic) [reply secrets/gcpssm/gcp_ssm.go]: Defined as const b95eb9d2fcd7a213bc19ef0ab427cde103615642

File: command/secrets/generate/secrets_generate.go (hunk 1 above)
Comment (zivkovicmilos) [reply command/secrets/generate/secrets_generate.go]: Yes

Comment (ZeljkoBenovic) [reply command/secrets/generate/secrets_generate.go]: Removed 981bd6d59617f854b8e2e991ee2f88a20e714e41

**Review by 0xAleksaOpacic — APPROVED** [review]
LGTM! Please add a link to doc pr in the description, also please resolve conflicts

**Pull request merged**